### PR TITLE
Mathew/visualizer improved ball and field drawing

### LINKED ISSUE #1081: Visualizer: Misc improvements to Ball and Field drawing
### Description of the task

* The field should show the physical boundary in addition to the out-of-bounds lines
* The goals should be highlighted on the field based on which team they belong to. Eg. the friendly goal should be highlighted green, enemy should be red
* Draw a cone between the ball and our net, showing the range of angles of attack

### Acceptance criteria

- [ ] Implement the above

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

* draw the physical boundary in addition to the out-of-bounds lines
* highlight the goals based on which team they belong to
* draw a cone between the ball and our net, showing the range of angles of attack
* Renamed the `gui` folder to `visualizer` for consistency
* Minor improvements to how colors are managed and shared in draw functions

### Testing Done

Manual testing, see screenshot

### Resolved Issues

resolves #1081 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by MathewMacDougall** [comment]
Screenshot from 2019-12-03 17-14-40

File: src/software/visualizer/drawing/navigator.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include "software/ai/navigator/navigator.h"
-#include "software/gui/drawing/draw_functions.h"
+#include "software/visualizer/drawing/draw_functions.h"
 
 // TODO: comment
```
Comment (MathewMacDougall) [inline src/software/visualizer/drawing/navigator.h:6]: I'm going to preemptively say this `TODO` was from before this PR and is being addressed in a separate set of changes I'm working on.

**Review by jonathanlew — APPROVED** [review]
Looks good. Does this support turning layers off? In the future, we'd want to turn on or off gradient descent, obstacles, etc

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Comment by MathewMacDougall** [comment]
@jonathanlew currently there is no support for layers and turning them on and off. I'll make a ticket for it for later.

**Pull request merged**